### PR TITLE
Auto owner set for API access

### PR DESCRIPTION
**Pull request by zefciu**
* Added owner field for domains/records
* Owner is auto-set when creating domains/records by API

### CODE REVIEW

File: powerdns/tests/test_ownership.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+from rest_framework.test import APIClient
+
+from powerdns.tests.utils import DomainFactory
+
+
+class TestOwnershipBase(TestCase):
+    """Base test class creating some users."""
+
+    def setUp(self):
+        self.user1 = User.objects.create_superuser(
```
Comment (andrzej-jankowski) [inline powerdns/tests/test_ownership.py]: Why not create a factory for users?

File: powerdns/tests/test_ownership.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            '/api/records/',
+            data={
+                'domain': '/api/domains/{}/'.format(self.domain.pk),
+                'type': 'CNAME',
+                'name': 'www.owned.example.com',
+                'content': 'blog.owned.example.com',
+                'owner': 'user2',
+            },
+        )
+        self.assertEqual(request.data['owner'], 'user2')
```
Comment (andrzej-jankowski) [inline powerdns/tests/test_ownership.py]: Your assertions are very similar in the whole file - maybe some custom assertion?

E.g.:
``_assert(self, expected_user, expected_mailbox_length)``

File: powerdns/utils.py [hunk 3]
```diff
@@ -17,6 +19,43 @@ class Meta:
         abstract = True
 
 
+class Owned(models.Model):
+    """Model that has an owner. This owner is set as default to the creator
+    of this model, but can be overridden."""
+
+    owner = models.ForeignKey('auth.User', null=True, blank=True)
```
Comment (andrzej-jankowski) [inline powerdns/utils.py]: Use ``settings.AUTH_USER_MODEL`` - will work with apps with custom user models.

File: powerdns/utils.py [hunk 4]
```diff
@@ -17,6 +19,43 @@ class Meta:
...
+
+    class Meta:
+        abstract = True
+
+    def email_owner(self, creator):
```
Comment (andrzej-jankowski) [inline powerdns/utils.py]: I am your package user and I don't want to send notifications in my project. How can I turn it off?

Comment (andrzej-jankowski) [inline powerdns/utils.py]: What happen if my custom user model does not have email field?

File: powerdns/utils.py [hunk 5]
```diff
@@ -17,6 +19,43 @@ class Meta:
...
+        """If the owner is different from the creator - notify the owner."""
+        if creator != self.owner:
+            subject_template, content_template = settings.OWNER_NOTIFICATIONS[
```
Comment (andrzej-jankowski) [inline powerdns/utils.py]: Information about ``settings.OWNER_NOTIFICATIONS`` should be in documentation.

File: powerdns/tests/test_ownership.py (hunk 1 above)
Comment (zefciu) [inline powerdns/tests/test_ownership.py]: It doesn't work for some reason. If I use a factory, I cannot login. I don't know why and I don't think there is point debugging this.

File: powerdns/utils.py (hunk 4 above)
Comment (zefciu) [inline powerdns/utils.py]: Do we really need to cover this? Especially that we plan to use it as microservice more and less as a django app.

**Comment by andrzej-jankowski** [comment]
:ok:

**Pull request merged**